google-cloud-spanner/lib/google/cloud/spanner/client.rb

```diff
@@ -1287,6 +1287,20 @@ def create_new_session
           Session.from_grpc grpc, @project.service
         end
 
+        ##
+        # @private
+        # Creates a batch of new session objects of size `session_count`.
+        def batch_create_new_sessions session_count
+          ensure_service!
+          resp = @project.service.batch_create_sessions \
```

Comment:
the response may have fewer sessions than requested in the RPC. u should check that and initiate another rpc (with requested_count - response_count) until requested_count == 0

Reply:
Will do, thank you for making us aware of this!!

Reply:
Done.